client/packages/invoices/src/Returns/api/hooks/document/useInsertSupplierReturn.ts

```diff
@@ -1,19 +1,23 @@
-import { useQueryClient, useMutation } from '@openmsupply-client/common';
+import {
+  useQueryClient,
+  useMutation,
+  useNavigate,
+  RouteBuilder,
+} from '@openmsupply-client/common';
 import { useReturnsApi } from '../utils/useReturnsApi';
+import { AppRoute } from 'packages/config/src';
 
 export const useInsertSupplierReturn = () => {
   const queryClient = useQueryClient();
-  // const navigate = useNavigate();
+  const navigate = useNavigate();
   const api = useReturnsApi();
   return useMutation(api.insertSupplierReturn, {
-    onSuccess: () => {
-      // TODO: redirect to details page
-      // onSuccess: invoiceNumber => {
-      // const route = RouteBuilder.create(AppRoute.Replenishment)
-      //   .addPart(AppRoute.InboundShipment)
-      //   .addPart(String(invoiceNumber))
-      //   .build();
-      // navigate(route, { replace: true });
+    onSuccess: invoiceNumber => {
+      const route = RouteBuilder.create(AppRoute.Replenishment)
+        .addPart(AppRoute.OutboundReturn)
+        .addPart(String(invoiceNumber))
+        .build();
+      navigate(route, { replace: true });
```

Comment:
This kind of weird to me, that the api hook navigates, I guess this was just following existing pattern ?

Reply:
Yeah this is what we've had elsewhere - happy to try changing it if we thing it's hiding/doing too much? But because we can call this insert both from the outbound shipment detail view and the returns list view, I found myself not minding the redirect logic only needing to be in one place...